test/matcher.cpp

```diff
@@ -113,132 +226,420 @@ TEST_CASE("Matcher throws if regex passed to `=~` or `!~` is invalid",
 	MatcherMockMatchable mock({{"AAAA", "12345"}});
 	Matcher m;
 
-	m.parse("AAAA =~ \"[[\"");
+	REQUIRE(m.parse("AAAA =~ \"[[\""));
 	REQUIRE_THROWS_AS(m.matches(&mock), MatcherException);
 
-	m.parse("AAAA !~ \"[[\"");
+	REQUIRE(m.parse("AAAA !~ \"[[\""));
 	REQUIRE_THROWS_AS(m.matches(&mock), MatcherException);
 }
 
 TEST_CASE("Operator `!~` checks if field doesn't match given regex",
 	"[Matcher]")
 {
-	MatcherMockMatchable mock({{"AAAA", "12345"}});
 	Matcher m;
 
-	m.parse("AAAA !~ \".\"");
-	REQUIRE_FALSE(m.matches(&mock));
+	SECTION("Works with strings") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA !~ \".\""));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA !~ \"123\""));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA !~ \"234\""));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA !~ \"45\""));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA !~ \"^12345$\""));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
+
+	SECTION("Converts numbers into strings and uses them as regexes") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA !~ 12345"));
+		REQUIRE_FALSE(m.matches(&mock));
 
-	m.parse("AAAA !~ \"123\"");
-	REQUIRE_FALSE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA !~ 1"));
+		REQUIRE_FALSE(m.matches(&mock));
 
-	m.parse("AAAA !~ \"234\"");
-	REQUIRE_FALSE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA !~ 45"));
+		REQUIRE_FALSE(m.matches(&mock));
 
-	m.parse("AAAA !~ \"45\"");
-	REQUIRE_FALSE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA !~ 9"));
+		REQUIRE(m.matches(&mock));
+	}
 
-	m.parse("AAAA !~ \"^12345$\"");
-	REQUIRE_FALSE(m.matches(&mock));
+	SECTION("Doesn't work with ranges") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA !~ 0:123456"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA !~ 12345:99999"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA !~ 0:12345"));
+		REQUIRE(m.matches(&mock));
+	}
 }
 
-TEST_CASE("Operator `#` checks if \"tags\" field contains given value",
+TEST_CASE("Operator `#` checks if space-separated list contains given value",
 	"[Matcher]")
 {
-	MatcherMockMatchable mock({{"tags", "foo bar baz quux"}});
 	Matcher m;
 
-	m.parse("tags # \"foo\"");
-	REQUIRE(m.matches(&mock));
+	SECTION("Works with strings") {
+		MatcherMockMatchable mock({{"tags", "foo bar baz quux"}});
```

Comment:
Given:
> "Operator `#` checks if space-separated list contains given value"

Might be nice to check if it checks for complete tags or also for parts of tags.

e.g. checking if `uu` matches (within `quux`):
```
REQUIRE(m.parse("tags # \"uu\""));
REQUIRE_FALSE(m.matches(&mock));
```